lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4272,6 +4272,20 @@ bool SPIRVToLLVM::transFPGAFunctionMetadata(SPIRVFunction *BF, Function *F) {
     MetadataVec.push_back(ConstantAsMetadata::get(getUInt32(M, Literals[1])));
     F->setMetadata(kSPIR2MD::LoopFuse, MDNode::get(*Context, MetadataVec));
   }
+  if (BF->hasDecorate(internal::DecorationMathOpDSPModeINTEL)) {
+    std::vector<SPIRVWord> Literals =
+        BF->getDecorationLiterals(internal::DecorationMathOpDSPModeINTEL);
+    assert(Literals.size() == 2 &&
+           "MathOpDSPModeINTEL decoration shall have 2 literals");
+    std::vector<Metadata *> MDDSPPref = {
+        ConstantAsMetadata::get(getUInt32(M, Literals[0]))};
+    F->setMetadata(kSPIR2MD::PreferDSP, MDNode::get(*Context, MDDSPPref));
```

Comment:
[`MDNode::get`](https://llvm.org/doxygen/classllvm_1_1MDNode.html#a7d10a7b9b7f40b04d27ed97c38ea1950) accepts `ArrayRef` as the second argument, which in turn [accepts reference to a single argument](https://llvm.org/doxygen/classllvm_1_1ArrayRef.html#a3b1f44186f9787d7ffacb54b62d6798c), you can avoid creating `std::vector` here and just pass `ConstantAsMetadata` as-is.